### PR TITLE
docs: split up contributing guides

### LINKED ISSUE #377: Docs: add "what are recipes/how to add recipe" page
Add a page to the docs and/or contribution guide on how to add individual recipes.

### PR DESCRIPTION
**Pull request by markmichon**
## Description
- Splits up the contributing guide(s) and moves it into the docs. 
- The existing `CONTRIBUTING.md` file has been slimmed down to link out to the necessary places.
- Adds a "how to contribute a recipe" guide.

## Related
- closes #377 

## Checklist

- [ ] I've added test coverage that shows my fix or feature works as expected.
- [x] I've updated or added documentation if required.
- [ ] I've included usage information in the description if CLI behavior was updated or added.
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

**Comment by markmichon** [comment]
@cfabianski I couldn't find much information on the recipe structure. Could use more clarification on the following:
- a list of supported types and sub_types
- how the "group" property is used for packages. Most are null, some are empty strings, and others seam to only be fore maven?
- the recommended generation format for UUIDs
- anything else that looks missing from the [recipes guide](https://github.com/Bearer/curio/blob/docs/contributing/docs/contributing/recipes.md)

**Comment by cfabianski** [comment]
> * a list of supported types and sub_types

- `external_service`
  - `third_party`

- `datastore`
  - `database`
  - `datalake`
  - `object_storage`
  - `search_engine`
  - `key_value_cache`
  - `flat_file`
  - `shared_folders`

- `internal_service`
  - `message_bus`

The subtype will be used for the dataflow

> * how the "group" property is used for packages. Most are null, some are empty strings, and others seam to only be fore maven?

The group is only for Java application yes as they are both needed to identify a package
https://maven.apache.org/guides/mini/guide-naming-conventions.html

> * the recommended generation format for UUIDs

```bash
# On Linux
uuidgen
# On MacOS [see](https://coderwall.com/p/ifxb-g/uuidgen-in-lowercase-on-mac-os-x)
uuidgen | tr "[:upper:]" "[:lower:]"
```

File: docs/docs.md [hunk 1]
```diff
@@ -17,16 +17,25 @@ Ready to dive in? Curio's [reports](/explanations/reports/) are your path to ana
 
 ## Explanations
 
+Explanations dive into the rational behind Curio and explain some of it's heavier concepts.
```
Comment (elsapet) [inline docs/docs.md]:
```suggestion
Explanations dive into the rational behind Curio and explain some of its heavier concepts.
```

**Review by elsapet — APPROVED** [review]
Looks good!

**Pull request merged**